src/main/java/com/boot/user/service/UserService.java

```diff
@@ -179,13 +184,19 @@ public GetUserDTO getUserByEmail(String email, boolean includeDetails, boolean i
 
     @Transactional
     public void deleteUserByEmail(String email) {
-        log.info("deleteUserByEmail - process started");
+        User user = userRepository.getUserByEmail(email);
 
-        if (userRepository.getUserByEmail(email) == null) {
+        if (user == null) {
             throw new EntityNotFoundException("Email: " + email + " not found in the Database!");
         }
 
-        userRepository.deleteByEmail(email);
+        ResponseEntity response = cartServiceClient.deleteCartByUserId(user.getId());
+        log.info("deleteCartByUserId - process started");
+
+        if (response.getStatusCode().name().equals(HttpStatus.OK.name())) {
+            userRepository.deleteUserByEmail(email);
+            log.info("deleteUserByEmail - process started");
```

Comment:
this is a process finish log..also add it outside the if condition

Reply:
i've changed it to "deleteUserByEmail - successfully deleted" because i wanted to be shure that " userRepository.deleteUserByEmail(email);" was run